File: update_management.py
import re

def update_file(filepath, pattern, replacement):
    content = open(filepath, 'r', encoding='utf-8').read()
    if 'especie_id' not in content:
        content = re.sub(pattern, replacement, content, count=1)
        open(filepath, 'w', encoding='utf-8').write(content)
        print(f"Updated {filepath}")
    else:
        print(f"Already updated {filepath}")

# LotManagement.tsx
update_file(
    'src/pages/Bovinocultura/LotManagement.tsx',
    r'(const payload = \{[\s\S]*?)(};)',
    r"\1  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \2"
)

# PastureManagement.tsx
update_file(
    'src/pages/Bovinocultura/PastureManagement.tsx',
    r'(const payload = \{[\s\S]*?)(};)',
    r"\1  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \2"
)

# ConfinementManagement.tsx
update_file(
    'src/pages/Bovinocultura/ConfinementManagement.tsx',
    r'(const payload = \{[\s\S]*?)(};)',
    r"\1  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \2"
)

File: update_management.js
const fs = require('fs');

function updateFile(filepath, pattern, replacement) {
    let content = fs.readFileSync(filepath, 'utf8');
    if (!content.includes('especie_id:')) {
        content = content.replace(pattern, replacement);
        fs.writeFileSync(filepath, content, 'utf8');
        console.log('Updated ' + filepath);
    } else {
        console.log('Already updated ' + filepath);
    }
}

// LotManagement.tsx
updateFile(
    'src/pages/Bovinocultura/LotManagement.tsx',
    /(const payload = \{[\s\S]*?)(};)/,
    "\  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \"
);

// PastureManagement.tsx
updateFile(
    'src/pages/Bovinocultura/PastureManagement.tsx',
    /(const payload = \{[\s\S]*?)(};)/,
    "\  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \"
);

// ConfinementManagement.tsx
updateFile(
    'src/pages/Bovinocultura/ConfinementManagement.tsx',
    /(const payload = \{[\s\S]*?)(};)/,
    "\  especie_id: 'bovino',\n      aptidao_id: 'corte',\n    \"
);
